### PR TITLE
Remove Context arguments from Aggregator.Checkpoint and Integrator.Process

### PR DESCRIPTION
**Pull request by jmacd**
None of these were used and these two methods are expected to perform only computation.

### CODE REVIEW

File: sdk/metric/sdk.go [hunk 1]
```diff
@@ -419,10 +418,8 @@ func (m *Accumulator) observeAsyncInstruments(ctx context.Context) int {
 	defer m.asyncLock.Unlock()
 
 	asyncCollected := 0
-	m.asyncContext = ctx
 
-	m.asyncInstruments.Run(context.Background(), m)
```
Comment (nilebox) [inline sdk/metric/sdk.go:424]: This is a change in behavior (a bugfix?), ideally should be done in a separate PR.

Comment (jmacd) [reply sdk/metric/sdk.go:424]: Done, thanks.
